Add eviction thresholds parameters (#42702)
kubernetes/website: content/en/docs/tasks/administer-cluster/kubelet-config-file.md

port: 20250
serializeImagePulls: false
evictionHard:
    memory.available:  "100Mi"
    nodefs.available:  "10%"
    nodefs.inodesFree: "5%"
    imagefs.available: "15%"
```

In this example, the kubelet is configured with the following settings:

> 1. address: The kubelet will serve on IP address 192.168.0.8.

Reviewer: ```suggestion
1. `address`: The kubelet will serve on IP address `192.168.0.8`.
```
similar below.

2. `port`: The kubelet will serve on port `20250`.
3. `serializeImagePulls`: Image pulls will be done in parallel.

> 4. `Eviction Thresholds`: The kubelet will evict Pods under the following conditions:

Reviewer 2: Is this true?

```suggestion
4. `evictionHard`: The kubelet will evict Pods under one of the following conditions:
```

Author: That definitely seems good.

>    - When the node's available memory drops below 200MiB.

Reviewer: ```suggestion
   - When the node's available memory drops below 100MiB.
```

   - When the node's main filesystem's available space is less than 10%.
   - When the image filesystem's available space is less than 15%.
   - When more than 95% of the node's main filesystem's inodes are in use.

{{< note >}}
In the example, by changing the default value of only one parameter for
evictionHard, the default values of other parameters will not be inherited and
will be set to zero. In order to provide custom values, you should provide all
the threshold values respectively.

> ---

Reviewer: Not sure if this is a common use in `note`.

Author: I think it's new.

Reviewer 3: I'm happy with it - I think it looks better than 2 consecutive notes.

The `imagefs` is an optional filesystem that container runtimes use to store container

> images and container writable layers.

Reviewer 2: These two lines do not have to be inside a note.

Author: Ok, Let me just change it.

Reviewer 2: Em. They are still in the note?

Author: My bad :sweat_smile: I forgot to push the code.

serializeImagePulls: false
evictionHard:
    memory.available:  "200Mi"
    nodefs.available:  "10%"
    imagefs.available: "15%"

>     nodefs.inodesFree: "5%"

Reviewer: Why not use the default 100Mi for memory.available?

```
	"memory.available":  "100Mi",
	"nodefs.available":  "10%",
	"nodefs.inodesFree": "5%",
	"imagefs.available": "15%",
```
Why not use alphabetic order?

Author: That seems good

serializeImagePulls: false
evictionHard:
    memory.available:  "200Mi"
    nodefs.available:  "10%"
    imagefs.available: "15%"
    nodefs.inodesFree: "5%"
```

In this example, the Kubelet is configured with the following settings:

1. address: The Kubelet will serve on IP address 192.168.0.8.
2. port: The Kubelet will serve on port 20250.
3. serializeImagePulls: Image pulls will be done in parallel.
4. Eviction Thresholds: The Kubelet will evict Pods under the following conditions:
   - When the node's available memory drops below 200MiB.
   - When the node's main filesystem's available space is less than 10%.
   - When the image filesystem's available space is less than 15%.

>    - When more than 95% of the node's main filesystem's inodes are in use.

Reviewer 3: This seems to say the same kind of thing as the following paragraph; do we need them both?

Author: Sorry my bad :sweat_smile:, raised this PR in a hurry.

Author: I removed the old text now.

the threshold values respectively.

> {{< /note >}}
> 
> {{< note >}}

Reviewer 3: ```suggestion

---

```

This produces one callout with a rule between the two points.
